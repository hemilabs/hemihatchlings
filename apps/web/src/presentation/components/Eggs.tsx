import React, { useState } from 'react'
import fireEgg from '../../assets/fire-egg.png'
import waterEgg from '../../assets/water-egg.png'
import grassEgg from '../../assets/grass-egg.png'
import chevron from '../../assets/chevron.svg'

function Eggs(): JSX.Element {
  const [active, setActive] = useState(0)

  const goPrevious = () => {
    setActive(active - 1)
  }

  const goNext = () => {
    setActive(active + 1)
  }

  return (
  <>
    <div className="flex flex-row justify-center gap-4 lg:gap-12">
      <button
        onClick={goPrevious}
        className='disabled:opacity-50 lg:hidden'
        disabled={active === 0}>
        <img src={chevron} className="w-8 rotate-180" alt="Go Left" />
      </button>

      <div className={active === 0 ? '' : 'hidden'}>
        <div className="p-8 border border-white bg-fire-yellow">
          <img src={fireEgg} alt="Fire Egg" />
        </div>
        <div className="font-monaco text-sm text-white mt-2
          sm:text-base md:text-lg md:mt-4">
          01
        </div>
      </div>

      <div className={`${active === 1 ? '' : 'hidden'} lg:block`}>
        <div className="p-8 border border-white bg-water-blue">
          <img src={waterEgg} alt="Water Egg" />
        </div>
        <div className="font-monaco text-sm text-white mt-2
          sm:text-base md:text-lg md:mt-4">
          02
        </div>
      </div>

      <div className={`${active === 2 ? '' : 'hidden'} lg:block`}>
        <div className="p-8 border border-white bg-grass-green">
          <img src={grassEgg} alt="Grass Egg" />
        </div>
        <div className="font-monaco text-sm text-white mt-2
          sm:text-base md:text-lg md:mt-4">
          03
        </div>
      </div>

      <button
        onClick={goNext}
        className='disabled:opacity-50 lg:hidden'
        disabled={active === 2}>
        <img src={chevron} className="w-8" alt="Go Right" />
      </button>
    </div>

    <div className="flex flex-col items-center">
      <h2 className="font-monaco font-normal text-white text-lg mt-8
        sm:text-xl md:text-2xl lg:text-3xl lg:mt-12 xl:text-4xl">
        FREE MINT + GAS FEES
      </h2>
    </div>
  </>
  )
}

export default Eggs
