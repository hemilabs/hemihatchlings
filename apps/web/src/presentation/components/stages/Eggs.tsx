import { useState } from 'react'
import fireEgg from '../../../assets/fire-egg.png'
import waterEgg from '../../../assets/water-egg.png'
import grassEgg from '../../../assets/grass-egg.png'
import airEgg from '../../../assets/air-egg.png'
import chevron from '../../../assets/chevron.svg'
import { ElementEnum } from '@hemihatchlings/shared'

interface EggsProps {
  hatchEgg: (element: ElementEnum) => void
}

function Eggs({ hatchEgg }:EggsProps): JSX.Element {
  const [active, setActive] = useState(0)

  const goPrevious = () => {
    setActive(active - 1)
  }

  const goNext = () => {
    setActive(active + 1)
  }

  return (
  <>
    <div className="flex flex-row justify-center gap-4 lg:gap-12">
      <button
        onClick={goPrevious}
        className='disabled:opacity-50 lg:hidden'
        disabled={active === 0}>
        <img src={chevron} className="w-32 rotate-180" alt="Go Left" />
      </button>

      <div className={active === 0 ? '' : 'hidden'}>
        <button onClick={() => hatchEgg(ElementEnum.Fire)}>
          <div className="border border-white">
            <img src={fireEgg} alt="Fire Egg" />
          </div>
          <div className="font-monaco text-sm text-white mt-2 text-left
            sm:text-base md:text-lg md:mt-4">
            01
          </div>
        </button>
      </div>

      <div className={`${active === 1 ? '' : 'hidden'} lg:block`}>
        <button onClick={() => hatchEgg(ElementEnum.Water)}>
          <div className="border border-white">
            <img src={waterEgg} alt="Water Egg" />
          </div>
          <div className="font-monaco text-sm text-white mt-2 text-left
            sm:text-base md:text-lg md:mt-4">
            02
          </div>
        </button>
      </div>

      <div className={`${active === 2 ? '' : 'hidden'} lg:block`}>
        <button onClick={() => hatchEgg(ElementEnum.Grass)}>
          <div className="border border-white">
            <img src={grassEgg} alt="Grass Egg" />
          </div>
          <div className="font-monaco text-sm text-white mt-2 text-left
            sm:text-base md:text-lg md:mt-4">
            03
          </div>
        </button>
      </div>

      <div className={`${active === 3 ? '' : 'hidden'} lg:block`}>
        <button onClick={() => hatchEgg(ElementEnum.Air)}>
          <div className="border border-white">
            <img src={airEgg} alt="Air Egg" />
          </div>
          <div className="font-monaco text-sm text-white mt-2 text-left
            sm:text-base md:text-lg md:mt-4">
            04
          </div>
        </button>
      </div>

      <button
        onClick={goNext}
        className='disabled:opacity-50 lg:hidden'
        disabled={active === 3}>
        <img src={chevron} className="w-32" alt="Go Right" />
      </button>
    </div>

    <div className="flex flex-col items-center">
      <h2 className="font-monaco font-normal text-white text-lg mt-8
        sm:text-xl md:text-2xl lg:text-3xl lg:mt-12 xl:text-4xl">
        FREE MINT + GAS FEES
      </h2>
    </div>
  </>
  )
}

export default Eggs
