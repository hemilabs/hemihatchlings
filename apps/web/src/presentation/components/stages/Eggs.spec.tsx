import { describe, it, expect} from 'vitest'
import { RenderResult, render } from '@testing-library/react'
import Eggs from './Eggs'

interface EggsProps {
  hatchEgg: (element: string) => void
}

const renderComponent = ({ hatchEgg }:EggsProps): RenderResult => {
    return render(
      <Eggs hatchEgg={hatchEgg} />
    )
  }

describe('src/presentation/Eggs', () => {
  it('renders correctly', () => {
    const { asFragment } = renderComponent({ hatchEgg: () => {}})

    expect(asFragment()).toMatchSnapshot()
  })
})
