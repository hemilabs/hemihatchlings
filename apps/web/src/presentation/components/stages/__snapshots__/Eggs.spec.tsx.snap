// Vitest Snapshot v1

exports[`src/presentation/Eggs > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row justify-center gap-4 lg:gap-12"
  >
    <button
      class="disabled:opacity-50 lg:hidden"
      disabled=""
    >
      <img
        alt="Go Left"
        class="w-32 rotate-180"
        src="/src/assets/chevron.svg"
      />
    </button>
    <div
      class=""
    >
      <button>
        <div
          class="border border-white transition duration-500 hover:scale-110"
        >
          <img
            alt="Fire Egg"
            src="./testnet/fire-egg.png"
          />
        </div>
        <div
          class="font-monaco text-sm text-white mt-2 text-left sm:text-base md:text-lg md:mt-4"
        >
          01
        </div>
      </button>
    </div>
    <div
      class="hidden lg:block"
    >
      <button>
        <div
          class="border border-white transition duration-500 hover:scale-110"
        >
          <img
            alt="Water Egg"
            src="./testnet/water-egg.png"
          />
        </div>
        <div
          class="font-monaco text-sm text-white mt-2 text-left sm:text-base md:text-lg md:mt-4"
        >
          02
        </div>
      </button>
    </div>
    <div
      class="hidden lg:block"
    >
      <button>
        <div
          class="border border-white transition duration-500 hover:scale-110"
        >
          <img
            alt="Grass Egg"
            src="./testnet/grass-egg.png"
          />
        </div>
        <div
          class="font-monaco text-sm text-white mt-2 text-left sm:text-base md:text-lg md:mt-4"
        >
          03
        </div>
      </button>
    </div>
    <button
      class="disabled:opacity-50 lg:hidden"
    >
      <img
        alt="Go Right"
        class="w-32"
        src="/src/assets/chevron.svg"
      />
    </button>
  </div>
  <div
    class="flex flex-col items-center"
  >
    <h2
      class="font-monaco font-normal text-white text-lg mt-8 sm:text-xl md:text-2xl lg:text-3xl lg:mt-12 xl:text-4xl"
    >
      FREE MINT + GAS FEES
    </h2>
  </div>
</DocumentFragment>
`;
